namespace.go

```diff
@@ -0,0 +1,105 @@
+package neo4j
+
+import (
+	"fmt"
+	"sort"
+)
+
+// Node represents each node in the trie
+type TrieNode struct {
+	children     map[rune]*TrieNode
+	isWord       bool
+	namespaceMap map[string]string
+}
+
+// Trie represents a trie and has a pointer to the root node
+type Trie struct {
+	root *TrieNode
+}
+
+// InitTrie will create a new Trie
+func InitTrie() *Trie {
+	result := &Trie{root: &TrieNode{
+		children: make(map[rune]*TrieNode),
+	}}
+	return result
+}
+
+// Insert will take in a word and add it to the trie
+func (t *Trie) Insert(word, mapping string) {
+	currentNode := t.root
+	for _, ch := range word {
+		if _, exists := currentNode.children[ch]; !exists {
+			newChild := &TrieNode{children: make(map[rune]*TrieNode), namespaceMap: map[string]string{}}
+			currentNode.children[ch] = newChild
+			currentNode = newChild
+		} else {
+			currentNode = currentNode.children[ch]
+		}
+	}
+	currentNode.namespaceMap[word] = mapping
+	currentNode.isWord = true
+}
+
+// Returns if there is any word in the trie that starts with the given prefix.
+func (t *Trie) PrefixStart(prefix string) bool {
+	currentNode := t.root
+	for _, ch := range prefix {
+		if _, ok := currentNode.children[ch]; !ok {
+			return false
+		}
+		currentNode = currentNode.children[ch]
+	}
+	return true
+}
+
+// Search will search if a word is in the trie
+func (t *Trie) Search(word string) (string, string, bool) {
+	currentNode := t.root
+	for _, ch := range word {
+		if _, ok := currentNode.children[ch]; !ok {
+			return word, fmt.Sprintf("No matching value for %s", word), false
+		}
+		currentNode = currentNode.children[ch]
+	}
+	if currentNode.isWord == true {
```

Comment:
You have to keep the last currentNode.mapping value. That's what you will return.